### ISSUE TITLE
charging and ev_connected is always Unknown in Home Assistant

### ISSUE THREAD

**Issue by magicus** [somlioy/amina_s#21]
This bug really refers to the code that was integrated into ZHC with https://github.com/Koenkk/zigbee-herdsman-converters/pull/8191, but I thought maybe we can figure out here what is wrong before opening a bug on that repo.

I really like how you added the two boolean `charging` and `ev_connected` values. That is kind of the essence of what I want to know to display in my home automation GUI. 

However, I can't get this to work properly. In the Z2M GUI, I can see the values properly displayed as "true" or "false", but when exported to Home Assistant, they always show up as `Unknown`. (Same goes for Alarms Active and Derated as well.)

Do you export the data to Home Assistant? Does this work for you?

**Comment by somlioy** [comment]
Yeah I also noticed that yesterday aswell... however the status text reports correctly to HA. 
I must examine this bug further

**Comment by magicus** [comment]
It seems like the boolean values should be sent as-is, and not as strings. I changed the new `binary` sections to:
```
            valueOn: [ true, 1],
            valueOff: [ false, 0],
```
and now it works.

**Comment by magicus** [comment]
I'm also a bit confused about the 
```
           valueOn: ['enable', 1],
            valueOff: ['disable', 0],
```
for e.g. `enable_offline`. 

This looks very nice in the Z2M GUI, but in Home Assistant this too fails to get a proper value. Instead the switch is presented as two separate buttons and the state is "Unknown". 

Changing this as well to `true` and `false` literals makes HA works, at the expense of a somewhat more boring GUI experience in Z2M.

**Comment by magicus** [comment]
I'm not sure why this is the case. One would think that it should have been possible to combine having a nicer Z2M GUI with proper HA functionality....

**Comment by somlioy** [comment]
I checked the value in the mqtt broker and the booleans are not sent as a string:
![image](https://github.com/user-attachments/assets/4ec92fde-82f6-442c-a000-aa3d30819b3a)

It might be something to do with the binary sensor in the HA-MQTT expecting "on" or "off", but somehow are able to translate 0 or 1?
https://www.home-assistant.io/integrations/binary_sensor.mqtt/#payload_on

**Comment by somlioy** [comment]
I checked the binary_sensor config which was created by mqtt auto discovery on the device, and it actually expected the values as "true" and "false". I changed those to true and false and it worked ...

mqtt topic:
homeassistant/binary_sensor/<unique_id>/charging
```javascript
{
  "payload_off": "false",
  "payload_on": "true",
}
``` 

Changed to

```javascript
{
  "payload_off": false,
  "payload_on": true,
}
``` 

![image](https://github.com/user-attachments/assets/b5fa1b15-2257-4962-9888-f19d4a18b1e8)

for the enable_offline it was set up as which seems to work for me
```javascript
{
  "payload_off": "disable",
  "payload_on": "enable",
}
```

**Comment by magicus** [comment]
So did you just change this MQTT topic manually, or could you convince Z2M to do this?

**Comment by somlioy** [comment]
Yes I changed the MQTT topic manually via MQTT Explorer, havent tested if this is retain tho.

Might be something useful in this article as well:
https://www.zigbee2mqtt.io/guide/usage/integrations/home_assistant.html
